data/blog/darwin-spelunking.mdx

```diff
@@ -0,0 +1,46 @@
+---
+title: 'Improving Tailscale via Apple’s open source'
+date: '2023-04-13'
+tags:
+  - 'oss'
+authors: ['mihai']
+images: ['/images/darwin-spelunking/social.png']
+summary: 'Spelunking through the Darwin kernel and userland helps Tailscale work better on macOS and iOS.'
+---
+
+# Improving Tailscale via Apple’s Open Source
+
```

Comment:
What do you think about boosting the lead here up a bit? I'm mostly thinking just inverting this paragraph. Maybe something like:

The ability to peek under the hood of the operating system can be a powerful tool for debugging. “Spelunking Apple's Open Source,” a recent blog post by Daniel Jalkut, struck a very familiar chord with me, reminding me of satisfying sessions poring through WebKit internals with my previous employer. I've been able to continue that pattern at Tailscale, now more focused on Darwin (the operating system at the core of macOS and iOS), its kernel, and its userland.

Just recently, Apple's open source came in handy when debugging before the Tailscale 1.36 release. We were investigating...